pkg/protoconv/time_test.go

```diff
@@ -77,3 +78,60 @@ func TestConvertMicroTSToProtobufTS(t *testing.T) {
 	assert.Equal(t, timestamp1.Seconds, convertedTimestamp1.Seconds)
 	assert.Equal(t, int32(123456000), convertedTimestamp1.Nanos)
 }
+
+func TestConvertTimestampToTimeOrDefault(t *testing.T) {
+	defaultGoTime := time.Now()
+	protoTime := &types.Timestamp{
+		Seconds: 1518046140,
+		Nanos:   123456789,
+	}
+	goTime := ConvertTimestampToTimeOrDefault(protoTime, defaultGoTime)
+	assert.Equal(t, goTime.Unix(), protoTime.GetSeconds())
+	assert.Equal(t, goTime.Nanosecond(), int(protoTime.GetNanos()))
+
+	goTime = ConvertTimestampToTimeOrDefault(nil, defaultGoTime)
+	assert.Equal(t, goTime.UTC(), defaultGoTime.UTC())
+
+	// Negative Nanos is not valid for Timestamp.
+	invalidProtoTime := &types.Timestamp{
+		Nanos: -1,
+	}
+	goTime = ConvertTimestampToTimeOrDefault(invalidProtoTime, defaultGoTime)
+	assert.Equal(t, goTime.UTC(), defaultGoTime.UTC())
+}
+
+func TestConvertTimeToTimestamp(t *testing.T) {
+	goTime := time.Now()
+	protoTime := ConvertTimeToTimestamp(goTime)
+	assert.Equal(t, goTime.Unix(), protoTime.GetSeconds())
+	assert.Equal(t, goTime.Nanosecond(), int(protoTime.GetNanos()))
+
+	invalidGoTime := goTime.AddDate(-10000, 0, 0)
+	protoTime = ConvertTimeToTimestamp(invalidGoTime)
+	assert.GreaterOrEqual(t, protoTime.GetSeconds(), goTime.Unix())
+	assert.GreaterOrEqual(t, int(protoTime.GetNanos()), goTime.Nanosecond())
```

Comment:
Why `GreaterOrEqual`?

Reply:
`ConvertTimeToTimestamp` will return the current time if conversion fails. We need to ensure that we have time after time marked at the beginning of the test. I have changed this because the assertions are wrong. 😒